src/mlpack/tests/softmax_regression_test.cpp

```diff
@@ -362,12 +362,8 @@ BOOST_AUTO_TEST_CASE(SoftmaxRegressionTrainTest)
 
   // This should be the same as the default parameters given by
   // SoftmaxRegression.
-  SoftmaxRegressionFunction srf(dataset, labels, 2, 0.0001, false);
-  L_BFGS<SoftmaxRegressionFunction> lbfgs(srf);
-  SoftmaxRegression<> sr(lbfgs);
-
+  SoftmaxRegression<> sr(dataset, labels, 2, 0.0001, false);
   SoftmaxRegression<> sr2(dataset.n_rows, 2);
-  sr2.Parameters() = srf.GetInitialPoint(); // Start from the same place.
```

Comment:
Ah yes, sorry, we should train the second object with an L_BFGS object:

```
L_BFGS lbfgs; // Use default options.
sr2.Train(dataset, labels, 2, lbfgs);
```

This way we test that passing an optimizer still gives us the same result.